magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala

```diff
@@ -571,6 +571,377 @@ class RecordsService(
+          entity(as[PutRecordsAspectRequest]) { requestData =>
+            parameters(
+              'merge.as[Boolean].?
+            ) { merge =>
+              withAuthDecision(
+                authClient,
+                AuthDecisionReqConfig("object/record/update")
+              ) { authDecision =>
+                val result = DB localTx { implicit session =>
+                  recordPersistence.putRecordsAspectById(
+                    tenantId,
+                    authDecision,
+                    requestData.recordIds,
+                    aspectId,
+                    requestData.data,
+                    userId,
+                    false,
+                    merge.getOrElse(false)
+                  ) match {
+                    case Success(result) =>
+                      complete(
+                        StatusCodes.OK,
+                        result
+                      )
+                    case Failure(ServerError(msg, statusCode)) =>
+                      complete(
+                        statusCode,
+                        ApiError(msg)
+                      )
+                    case Failure(e) =>
+                      complete(
+                        StatusCodes.InternalServerError,
+                        ApiError(e.getMessage)
+                      )
+                  }
+                }
+                webHookActor ! WebHookActor
+                  .Process()
+                result
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+    * @apiGroup Registry Record Aspects
+    * @api {delete} /v0/registry/records/aspectArrayItems/:aspectId Remove items from records' aspect data
+    *
+    * @apiDescription  this API goes through the aspect data that is specified by aspectId of a list of registry records
+    * and delete items from the array that is located by the jsonPath.
+    * If the aspect doesn't exist for a record or the array can't be located with the jsonPath string or the value located with the jsonPath string is `null`,
+    * the operation will be skipped without throwing error. `0` will returned as eventId for this case.
+    * If the json data that is located by the jsonPath string exists and is not an Array or null, an error will be thrown and 400 code will be responded.
+    *
+    * @apiParam (path) {string} aspectId the id of the aspect to be updated
+    * @apiParam (body) {string[]} recordIds a list of record IDs of records to be patched
+    * @apiParam (body) {string} jsonPath the jsonPath string that is used to locate the json array in the record aspect data
+    * @apiParam (body) {any[]} items a list of items to be removed from the located array. The type of the items can be either string or number.
+    *
+    * @apiParamExample {json} Request-Example
+    * {
+    *   "recordIds": ["dsd-sds-xsds-22", "sds-sdds-2334-dds-34", "sdds-3439-34334343"],
+    *   "aspectId": "access-control",
```

Comment:
```suggestion
    *   "aspectId": "access-control",
```

Should `aspectId` be in the example request body?

Reply:
Thanks it was mistake in API docs --- the aspectId is supplied via path segment rather than body